### ISSUE TITLE
Atum / Cassiopeia: User session time limit causes involuntarily logout

### ISSUE THREAD

**Issue by mbeganyi-a11y** [joomla/joomla-cms#48419]
### Steps to reproduce the issue

**NOTE**: Accessibility backlog issue and work-in-progress. This issue has been opened to account for all the issues flagged in the initial [Milestone 1a Audit Report](https://accessibility-project.joomla.org/results/Milestone-1a/milestone1/deliverable1/wcag-report). If this is a duplicate issue, it will be closed by the end of September 2026. 

Milestone 1a   finding ID | M1a-004
-- | --
Backlog   priority | High
Severity —   carried from Milestone 1a | Serious
Implementation   effort | Medium
Target   milestone(s) | MS2a, MS3a,   MS3b, MS7a, MS7b, MS7c, MS8a, MS9a, MS10
WCAG 2.2   success criterion | 2.2.1 Timing Adjustable
Responsibility   tag(s) from report | Design,   Development
Affected   Joomla area/component | https://accessibility-project.joomla.org/results/Milestone-1a/milestone1/deliverable1/wcag-report#c81fc029-4b10-400e-86a1-ccb8814587ee
Affected user   journey(s) | 01 – Backend   Template (Atum); 05 – MFA Setup - Method / Methods Views; 06 – Media Manager   (com_media); 07 – Installer Wizard (com_installer/Install); 08 – Backend List   View Pattern; 09 – Backend Edit Form Pattern; 11 – Backend Options including   Permissions; 12 – Update Joomla
Source   report/reference | Milestone 1a   WCAG audit report
STF reporting   reference | To be   assigned
Issue owner | To be   assigned
Status | Triage

<p class=MsoNormal>Audit finding<o:p></o:p></p>

<p class=MsoNormal>User session expires without warning<o:p></o:p></p>

<p class=MsoNormal>User impact<o:p></o:p></p>

<p class=MsoNormal>Users who need more time may lose progress or be unable to
complete the task before the time limit expires.<o:p></o:p></p>

<p class=MsoNormal>Remediation plan<o:p></o:p></p>

<p class=MsoNormal>Implement a component-level and template-level fix that allows for session expiry limit to be extended or dismiss the expiry and continue.<o:p></o:p></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'>Steps to reproduce</b><o:p></o:p></p>

<p class=MsoNormal>1. Open an area of Joomla that requires logging in.<o:p></o:p></p>

<p class=MsoNormal>2. Note the session timeout behaviour by idling the session
for 15 minutes.<o:p></o:p></p>

<p class=MsoNormal>3. Wait until the warning or timeout occurs.<o:p></o:p></p>

<p class=MsoNormal>4. Attempt to extend the session and continue the task.<o:p></o:p></p>

<p class=MsoNormal>5. Observe that the user is automatically signed out without
warning.<o:p></o:p></p>

</body>

</html>

### Version

6.1

### Expected result

Users can do one of the following actions:

- Turn off: The user is allowed to turn off the time limit before encountering it; or
- Adjust: The user is allowed to adjust the time limit before encountering it over a wide range that is at least ten times the length of the default setting; or
- Extend: The user is warned before time expires and given at least 20 seconds to extend the time limit with a simple action (for example, "press the space bar"), and the user is allowed to extend the time limit at least ten times.

Alternatively, extend the session timeout to over 20 hours.

### Actual result

User is automatically logged out after an idle period of 15 minutes.

### What device were you using? (eg, phone, laptop)

Desktop web

### What web browser were you using? (eg, Chrome, Safari)

Chrome, Firefox, Edge, Safari

### Assistive Technology?

N/A

**Comment by brianteeman** [comment]
Again not 100% true

Any page that has user interaction has behaviour.keepalive

**Comment by mbeganyi-a11y** [comment]
I'll need to investigate this also as users can set timeouts in global settings.

**Comment by rdeutz** [comment]
As Brian said, if you edit an article for example you are not looged out. We need to find a good balance between security and UX here. 15 minutes might be to short but 20 hours is way too long. Further more admins can set the seesion time for a site in respecting the  used group needs.

I would close the issue.

**Comment by mbeganyi-a11y** [comment]
Yeah I agree. Setting timeouts in the global settings is acceptable, since you're giving users the option to extend at the origin. I suppose the only thing worth considering is a session timeout dialog that allows users to extend the timeout. This avoids being logged out without warning, but again, in my opinion, is something to consider implementing and not a must.

**Comment by brianteeman** [comment]
There are extensions available for that
